src/ServiceControl.Transports/TransportCustomization.cs

```diff
@@ -5,19 +5,50 @@
     using System.Threading.Tasks;
     using NServiceBus;
+    using NServiceBus.Features;
     using NServiceBus.Raw;
     using NServiceBus.Transport;
 
     public abstract class TransportCustomization
     {
-        public abstract void CustomizeSendOnlyEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
+        protected abstract void CustomizeTransportSpecificServiceControlEndpointSettings(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
 
-        public abstract void CustomizeServiceControlEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
+        protected abstract void CustomizeTransportSpecificSendOnlyEndpointSettings(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
 
-        public abstract void CustomizeMonitoringEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
+        protected abstract void CustomizeTransportSpecificMonitoringEndpointSettings(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
 
         public abstract void CustomizeForReturnToSenderIngestion(RawEndpointConfiguration endpointConfiguration, TransportSettings transportSettings);
 
+        public void CustomizeServiceControlEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings)
+        {
+            ConfigureDefaultEndpointSettings(endpointConfiguration, transportSettings);
+            CustomizeTransportSpecificServiceControlEndpointSettings(endpointConfiguration, transportSettings);
+        }
+
+        public void CustomizeSendOnlyEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings)
+        {
+            ConfigureDefaultEndpointSettings(endpointConfiguration, transportSettings);
+            CustomizeTransportSpecificSendOnlyEndpointSettings(endpointConfiguration, transportSettings);
+        }
+
+        public void CustomizeMonitoringEndpoint(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings)
+        {
+            ConfigureDefaultEndpointSettings(endpointConfiguration, transportSettings);
+            CustomizeTransportSpecificMonitoringEndpointSettings(endpointConfiguration, transportSettings);
+        }
+
+        protected void ConfigureDefaultEndpointSettings(EndpointConfiguration endpointConfiguration, TransportSettings transportSettings)
+        {
+            endpointConfiguration.DisableFeature<Audit>();
+            endpointConfiguration.DisableFeature<AutoSubscribe>();
+            endpointConfiguration.DisableFeature<Outbox>();
+            endpointConfiguration.DisableFeature<Sagas>();
+            endpointConfiguration.DisableFeature<TimeoutManager>();
+            endpointConfiguration.SendFailedMessagesTo(transportSettings.ErrorQueue);
+
+            endpointConfiguration.UsePersistence<InMemoryPersistence>();
```

Comment:
The main instance needs to use the cached raven persistence

Reply:
Once I have the tests passing I'll move the config out of the main code and switch to configuring NSB properly here